h/services/nipsa.py

```diff
@@ -23,14 +22,13 @@ def flagged_userids(self):
 
         :rtype: set of unicode strings
         """
-        if self._flagged_userids is None:
-            query = self.session.query(User).filter_by(nipsa=True)
-            self._flagged_userids = set([u.userid for u in query])
-        return self._flagged_userids
+        query = self.session.query(User).filter_by(nipsa=True)
```

Comment:
This query is still inefficient AFAIK since `user.nipsa` is not indexed, but it doesn't matter since it is only used by the `/admin/nipsa` view following the change to `is_flagged` below.

Reply:
You are correct it does not appear to be indexed and this may be slow for the admin view but I suppose we aren't overly concerned about that use case.